### PR TITLE
swaynag: adds option to set wayland shell layer

### PR DESCRIPTION
**Pull request by Jamedjo**
Uses --layer/-y set to overlay|top|bottom|background

Closes https://github.com/swaywm/sway/issues/4939

### CODE REVIEW

**Review by fluix-dev — APPROVED** [review]

File: swaynag/types.c [hunk 1]
```diff
@@ -100,6 +101,10 @@ void swaynag_type_merge(struct swaynag_type *dest, struct swaynag_type *src) {
 		dest->anchors = src->anchors;
 	}
 
+	if (src->layer > 0) {
```
Comment (emersion) [inline swaynag/types.c:104]: `ZWLR_LAYER_SHELL_V1_LAYER_BACKGROUND == 0` so we can't use this as a guard value.

File: swaynag/swaynag.c [hunk 2]
```diff
@@ -476,7 +476,8 @@ void swaynag_setup(struct swaynag *swaynag) {
 	swaynag->layer_surface = zwlr_layer_shell_v1_get_layer_surface(
 			swaynag->layer_shell, swaynag->surface,
 			swaynag->output ? swaynag->output->wl_output : NULL,
-			ZWLR_LAYER_SHELL_V1_LAYER_TOP, "swaynag");
+			swaynag->type->layer ? swaynag->type->layer : ZWLR_LAYER_SHELL_V1_LAYER_TOP,
```
Comment (emersion) [inline swaynag/swaynag.c:479]: Ditto

File: swaynag/types.c (hunk 1 above)
Comment (Jamedjo) [reply swaynag/types.c]: > `ZWLR_LAYER_SHELL_V1_LAYER_BACKGROUND == 0` so we can't use this as a guard value.

Thanks for spotting this, I've updated it to use a `int32_t` with an initial value of `-1`.

We could alternatively remove support for the `background` setting, as I'm unaware of the use case. From my testing the nag bar appears behind the desktop background, but if a smaller bg is centred on the page it does show and looks similar to using a value of `bottom`.

**Comment by emersion** [comment]
Hm, CI is failing for some reason -- maybe rebase against master to see if that fixes it?

File: include/swaynag/types.h [hunk 3]
```diff
@@ -7,6 +7,7 @@ struct swaynag_type {
 	char *font;
 	char *output;
 	uint32_t anchors;
+	int32_t layer;
```
Comment (emersion) [inline include/swaynag/types.h:10]: Nit: ideally add a small `// enum zwlr_layer_shell_v1_layer or -1 if unset` comment to make it clear what the values can be.

**Review by emersion — APPROVED** [review]
LGTM, thanks!

**Pull request merged**